numpy/core/tests/test_numeric.py

```diff
@@ -954,8 +954,7 @@ class NIterError(Exception):
 
 class TestFromiter:
     def makegen(self):
-        for x in range(24):
-            yield x**2
+        yield from (x**2 for x in range(24))
```

Comment:
Nit:
```suggestion
        return (x**2 for x in range(24))
```

Reply:
Then `makegen` will not return a generator. Can `np.fromiter(val)` us a non-generator for `val`?

Reply:
>will not return a generator.

The last line of the function is `return <a generator>`, so I don't believe you.

Reply:
Thanks for suggestion. And yes, it's a generator.